eclair-core/src/main/scala/fr/acinq/eclair/transactions/CommitmentSpec.scala

```diff
@@ -62,54 +61,62 @@ object CommitmentSpec {
     case _ => false
   }
 
-  def addHtlc(spec: CommitmentSpec, direction: Direction, update: UpdateAddHtlc): CommitmentSpec = {
-    val htlc = direction match {
-      case IN => IncomingHtlc(update)
-      case OUT => OutgoingHtlc(update)
-    }
-    direction match {
-      case OUT => spec.copy(toLocal = spec.toLocal - htlc.add.amountMsat, htlcs = spec.htlcs + htlc)
-      case IN => spec.copy(toRemote = spec.toRemote - htlc.add.amountMsat, htlcs = spec.htlcs + htlc)
+  def addHtlc(spec: CommitmentSpec, directedHtlc: DirectedHtlc): CommitmentSpec = {
+    directedHtlc match {
+      case OutgoingHtlc(add) => spec.copy(toLocal = spec.toLocal - add.amountMsat, htlcs = spec.htlcs + directedHtlc)
+      case IncomingHtlc(add) => spec.copy(toRemote = spec.toRemote - add.amountMsat, htlcs = spec.htlcs + directedHtlc)
     }
   }
 
   // OUT means we are sending an UpdateFulfillHtlc message which means that we are fulfilling an HTLC that they sent
-  def fulfillHtlc(spec: CommitmentSpec, direction: Direction, htlcId: Long): CommitmentSpec = {
-    spec.findHtlcById(htlcId, direction.opposite) match {
-      case Some(htlc) if direction == OUT => spec.copy(toLocal = spec.toLocal + htlc.add.amountMsat, htlcs = spec.htlcs - htlc)
-      case Some(htlc) if direction == IN => spec.copy(toRemote = spec.toRemote + htlc.add.amountMsat, htlcs = spec.htlcs - htlc)
+  def fulfillOutgoingHtlc(spec: CommitmentSpec, htlcId: Long): CommitmentSpec = {
```

Comment:
I think the naming becomes counter-intuitive with this refactoring.
It made sense to say `fulfillHtlc(spec, OUT, id)` to mean that you were the one sending the fulfill (and `fulfillHtlc(spec, IN, id)` when you were the one receiving the fulfill).

Now it's really weird because we're saying `fulfillOutgoingHtlc` for an incoming HTLC, and `fulfillIncomingHtlc` for an outgoing HTLC...I would reverse that: `fulfillOutgoingHtlc` should fulfill an HTLC you sent, and `fulfillIncomingHtlc` should fulfill one you received.

It's the same for the `failXXX` functions below.

Reply:
Agree, this file is a mess, surely we can do better.